qa/tasks/cephfs/test_volumes.py

```diff
@@ -7000,6 +7000,11 @@ def test_subvolume_snapshot_clone_cancel_pending(self):
         # snapshot subvolume
         self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
 
+        # Disable the snapshot_clone_no_wait config option
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_no_wait', False)
+        threads_available = self.config_get('mgr', 'mgr/volumes/snapshot_clone_no_wait')
+        self.assertEqual(threads_available, 'false')
```

Comment:
This is the culprit for the test failure. After this test is executed, the config `snapshot_clone_no_wait` is flipped to `false` (default is true). Your test `test_subvolume_snapshot_clone_with_no_wait_enabled` is being executed with `false` where as your test expects the default value which is `true`. 

Your test should either explicitly set the config to `true` or validate and set correctly if it's not before proceeding further.

Reply:
The config are retained across tests. So as a practice, it's important to revert the configs to default at the end of the tests. The same is applicable to `max_concurrent_clones` and `snapshot_delay` in your tests.

Reply:
Here is the proof from the teuthology logs that this test ran first

```
kotresh:Downloads$ grep ":>" teuthology.log.1 | grep -E "Starting|Ended" | grep -E "test_subvolume_snapshot_clone_cancel_pending|test_subvolume_snapshot_clone_with_no_wait_enabled"
2024-02-08T15:08:37.378 DEBUG:teuthology.orchestra.run.smithi070:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Starting test tasks.cephfs.test_volumes.TestSubvolumeSnapshotClones.test_subvolume_snapshot_clone_cancel_pending'
2024-02-08T15:10:01.547 DEBUG:teuthology.orchestra.run.smithi070:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Ended test tasks.cephfs.test_volumes.TestSubvolumeSnapshotClones.test_subvolume_snapshot_clone_cancel_pending'
2024-02-08T15:20:26.789 DEBUG:teuthology.orchestra.run.smithi070:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Starting test tasks.cephfs.test_volumes.TestSubvolumeSnapshotClones.test_subvolume_snapshot_clone_with_no_wait_enabled'
2024-02-08T15:20:59.218 DEBUG:teuthology.orchestra.run.smithi070:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Ended test tasks.cephfs.test_volumes.TestSubvolumeSnapshotClones.test_subvolume_snapshot_clone_with_no_wait_enabled'

```

Also note from mgr logs that the config flipped to `false` after that test is run. Check the timing of test above.

```
kotresh:Downloads$ zgrep "snapshot_clone_no_wait =" ceph-mgr.y.log.gz 
...
...
2024-02-08T14:55:08.563+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T14:55:38.210+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T14:56:30.387+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T14:57:01.202+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T14:57:57.215+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T15:08:13.975+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T15:08:36.389+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = True
2024-02-08T15:09:13.930+0000 7f29f8bf5640 20 mgr update_kv_data  set config/mgr/mgr/volumes/snapshot_clone_no_wait = false
2024-02-08T15:09:13.940+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:10:01.542+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:20:45.004+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:20:45.606+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:20:58.910+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:20:59.212+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False
2024-02-08T15:21:28.064+0000 7f297d153640  0 [volumes DEBUG root]  mgr option snapshot_clone_no_wait = False

```

Reply:
Thanks for looking into it @kotreshhr . I did look into logs and found the same. I scheduled a run with a change where I only enabled the config option at the end of the tests but I saw tests were failing in that case too. So, I added the lines in the test for enabling/disabling the config option `snapshot_clone_no_wait` at the start of the tests and checking their correctness. AFAIR, the test was initially designed this way only but need to remove this enabling and checking of this config option `snapshot_clone_no_wait` in `test_subvolume_snapshot_clone_with_no_wait_enabled`  due to any comment/suggestion and the tests were passing individually in vstart. I forgot that the config options are retained through out the tests and the order of tests teuthology run may not be same as they are written.

Reply:
Here's the link for successful run: https://pulpito.ceph.com/neesingh-2024-02-13_07:26:50-fs:volumes-wip-neesingh-testing1-20240208-distro-default-smithi/